File: python/src/logica_programacion/ejercicio2.py
# Ejercicio #2:


def es_palindromo(palabra):
    """
    Verifica si la cadena proporcionada es un palíndromo.

    :param palabra: (str), La cadena a verificar.
    :return: (bool), True si la cadena es un palíndromo, False de lo contrario.
    """

    if isinstance(palabra, str):
        palabra = palabra.lower()
        return palabra == ''.join(reversed(palabra))

    return False


# print(es_palindromo("Ana"))

File: typescript/src/ejercicio2.ts
// Ejercicio #2:


export default function esPalindromo(palabra:string):boolean {
    /**
     * Verifica si la cadena proporcionada es un palíndromo.
     * @param palabra La cadena a verificar.
     * @return true si la cadena es un palíndromo, false de lo contrario.
     */

    palabra = palabra.toLowerCase()
    return palabra === palabra.split('').reverse().join('')
}


// console.log(esPalindromo("Ana"))
